File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, l: List[int], k: int) -> int:
        s = 0
        N = len(l)
        res = 0
        for i in range(N-k, len(l)):
            res += l[i]
        s = res
        for i in range(k):
            s -= l[N-k+i]
            s += l[i]
            res = max(s, res)
        return res

File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] l, int k) {
        int s = 0;
        int N = l.length;
        int res = 0;
        for(int i = N-k; i < N; i++){
                res += l[i];
            }
        s = res;
        for(int i = 0; i < k; i++){
                s -= l[N-k+i];
                s += l[i];
                res = Math.max(s, res);
            }
        return res;
    }
}